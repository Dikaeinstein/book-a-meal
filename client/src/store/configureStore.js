import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
