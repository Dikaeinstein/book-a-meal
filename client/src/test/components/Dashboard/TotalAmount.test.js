import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedTotalAmount, { TotalAmount }
  from '../../../components/Dashboard/TotalAmount';


describe('<totalAmount />', () => {
  it('should call fetAllOrders when component mounts', () => {
    const fetchTotalAmount = jest.fn();
    shallow(<TotalAmount
      totalAmount={4}
      userRole="superAdmin"
      fetchTotalAmount={fetchTotalAmount}
      fetchCatererTotalAmount={jest.fn()}
    />);
    expect(fetchTotalAmount).toHaveBeenCalled();
  });
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({
      user: { data: { role: 'superAdmin' } },
      dashboard: { totalAmount: 4 },
    });
    const wrapper = shallow(<ConnectedTotalAmount store={store} />);
    expect(wrapper.props().userRole).toEqual('superAdmin');
    expect(wrapper.props().totalAmount).toEqual(4);
  });
  it('should match snapshot', () => {
    const tree = shallow(<TotalAmount
      totalAmount={4}
      userRole="caterer"
      fetchTotalAmount={jest.fn()}
      fetchCatererTotalAmount={jest.fn()}
    />);
    expect(tree).toMatchSnapshot();
  });
});
