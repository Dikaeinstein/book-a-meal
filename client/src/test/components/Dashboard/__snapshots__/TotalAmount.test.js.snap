// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<totalAmount /> should match snapshot 1`] = `
<div
  className="dashboard-item"
>
  <p
    className="font-weight-bold"
  >
    Total Amount
  </p>
  <h2
    className="amount font-weight-bold"
  >
    â‚¦ 
    4
  </h2>
</div>
`;
